.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,28 @@
+name: "Setup build environment"
```

Comment:
nit: why setup-build/action.yml instead of setupBuild.yml?

Reply:
I'm pretty sure I tried that and it didn't work. Actions need a file called [action.yml](https://docs.github.com/en/actions/creating-actions/about-custom-actions#types-of-actions). If you're referring to why the action isn't named in camel case, I think this is just the conventional style for github actions